File: solutions/scala/raindrops/1/src/main/scala/Raindrops.scala
object Raindrops {
  private val Sounds =
    List(3 -> "Pling", 5 -> "Plang", 7 -> "Plong")

  def convert(n: Int): String =
    val sounds =
      Sounds
        .collect { case (factor, sound) if n % factor == 0 => sound }
        .mkString

    if sounds.isEmpty then n.toString else sounds
}

File: solutions/fsharp/raindrops/1/Raindrops.fs
module Raindrops

let convert (number: int): string =
    let sounds = [ (3, "Pling"); (5, "Plang"); (7, "Plong") ]

    let result =
        sounds
        |> List.choose (fun (factor, sound) ->
            if number % factor = 0 then Some sound else None)
        |> String.concat ""

    if result = "" then string number else result
